### PR TITLE
Review Comments

### PR DESCRIPTION
**Pull request by egekorkan**
Followed http://astrofrog.github.io/blog/2013/04/10/how-to-conduct-a-full-code-review-on-github/ 

We can use this for discussions

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Sep 25, 2020, 12:00 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Frawcdn.githack.com%2Fw3c%2Fwot-scripting-api%2F9816307eabcf8302b02af406f2bbc86ad0606d10%2Findex.html%3FisPreview%3Dtrue)

```
[33m🕵️‍♀️  That doesn't seem to be a ReSpec document. Please check manually:[39m [36m[Function: url][39m
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/wot-scripting-api%23247.)._
</details>

### CODE REVIEW

**Comment by egekorkan** [comment]
There are no review comments in here yet, will comment accordingly.

**Comment by zolkis** [comment]
I created #248 for how to do this :).
Feel free to use it, or delete your PR branch (closing this PR) and create your similar PR like the one above (then I will remove my PR).

File: index.html [hunk 1]
```diff
@@ -0,0 +1,3644 @@
...
+      that is encapsulated by implementations.
+    </p>
+  </section>
+
+  </section> <!-- Handling interaction data -->
+
+  <section data-dfn-for="ConsumedThing">
+    <h2>The <dfn>ConsumedThing</dfn> interface</h2>
+    <p>
+      Represents a client API to operate a <a>Thing</a>. Belongs to the
```
Comment (egekorkan) [inline index.html:995]: We should refrain from using client and server terms

File: index.html [hunk 2]
```diff
@@ -0,0 +1,3644 @@
...
+          return value;
+        };
+      </pre>
+    </section> <!-- Examples -->
+  </section> <!-- ConsumedThing -->
+
+  <section data-dfn-for="ExposedThing">
+    <h2>The <dfn>ExposedThing</dfn> interface</h2>
+    <p>
+      The {{ExposedThing}} interface is the server API to operate the <a>Thing</a> that allows defining request handlers, <a>Property</a>, <a>Action</a>, and <a>Event</a> interactions.
```
Comment (egekorkan) [inline index.html:1734]: We should refrain from using client and server terms

File: index.html [hunk 3]
```diff
@@ -0,0 +1,3644 @@
...
+        The {{ExposedThing}} interface extends {{ConsumedThing}}. It
+        is constructed from a full or partial {{ThingDescription}} object.
+      </p>
+      <p class="note">
+         Note that an existing {{ThingDescription}} object can be optionally modified (for instance by adding or removing elements on its |properties|, |actions| and |events| internal properties) and the resulting object can used for constructing an
+         {{ExposedThing}} object. This is the current way of adding and
+         removing <a>Property</a>, <a>Action</a> and <a>Event</a> definitions, as illustrated in the <a href="#exposedthing-examples">examples</a>.
+      </p>
+      <p class="note">
+        Before invoking <a href="#dom-exposedthing-expose">expose()</a>, the {{ExposedThing}} object does not serve any requests. This allows first constructing {{ExposedThing}} and then initialize its <a>Properties</a> and service handlers before starting serving requests.
```
Comment (egekorkan) [inline index.html:1790]: and then initializing

File: index.html [hunk 4]
```diff
@@ -0,0 +1,3644 @@
...
+          <li>
+            Return |thing|.
+          </li>
+        </ol>
+      </div>
+    </section>
+    <section>
+      <h3>Methods inherited from {{ConsumedThing}}</h3>
+      <p>
+        The <code>readProperty()</code>, <code>readMultipleProperties()</code>, <code>readAllProperties()</code>, <code>writeProperty()</code>, <code>writeMultipleProperties()</code>, <code>writeAllProperties()</code> methods have the same algorithmic steps as described in <a href="#the-consumedthing-interface"><code>ConsumedThing</code></a>, with the difference that making a request to the underlying platform MAY be implemented with local methods or libraries and don't necessarily need to involve network operations.
```
Comment (egekorkan) [inline index.html:1817]: doesn't necessarily need to involve

File: index.html [hunk 5]
```diff
@@ -0,0 +1,3644 @@
...
+        Throws on error.
+        Returns a reference to |this| object for supporting chaining.
+      </p>
+      <p>
+         The |handler| callback function should implement reading a
+         <a>Property</a> and resolve with an {{InteractionOutput}} object or
+         reject with an error.
+      </p>
+      <p>
+        There MUST be at most one handler for any given <a>Property</a>, so newly added handlers MUST replace the previous handlers. If no handler is initialized for any given <a>Property</a>, implementations SHOULD implement a default property read handler based on the <a>Thing Description</a>.
```
Comment (egekorkan) [inline index.html:2028]: at most one observeproperty handler

File: index.html [hunk 6]
```diff
@@ -0,0 +1,3644 @@
...
+        received for unobserving the specified <a>Property</a> matched by |name|.
+        Throws on error.
+        Returns a reference to |this| object for supporting chaining.
+      </p>
+      <p>
+        The |handler| callback function should implement what to do when an
+        unobserve request is received by the implementation.
+      </p>
+      <p>
+        There MUST be at most one handler for any given <a>Property</a>, so newly added handlers MUST replace the previous handlers. If no handler is initialized for any given <a>Property</a>, implementations SHOULD implement a default handler based on the <a>Thing Description</a>.
```
Comment (egekorkan) [inline index.html:2136]: at most one unobserveproperty handler

File: index.html [hunk 7]
```diff
@@ -0,0 +1,3644 @@
...
+          <li>
+            If an <a>Event</a> with the name |name| is not found,
+            [= exception/throw =] a {{NotFoundError}} and abort these steps.
+          </li>
+          <li>
+            Set the <a>internal slot</a> ||subscribeHandler|| on |interaction|
+            to |handler|.
+          </li>
+          <li>
+            Return `this`.
```
Comment (egekorkan) [inline index.html:2499]: `this` is not very clear, if needed why  other handlers don't do it?

File: index.html [hunk 8]
```diff
@@ -0,0 +1,3644 @@
...
+            Otherwise resolve |promise| and abort these steps.
+          </li>
+        </ol>
+      </div>
+    </section>
+
+    <section>
+      <h2>ExposedThing Examples</h2>
+      <p>
+        The next example illustrates how to create an {{ExposedThing}} based on a partial <a>TD</a> object constructed beforehands.
```
Comment (egekorkan) [inline index.html:2818]: beforehand

File: index.html [hunk 9]
```diff
@@ -0,0 +1,3644 @@
...
+              reset: {
+                description: "Reset the temperature sensor",
+                input: {
+                  temperature: temperatureValueDefinition
+                },
+                output: null,
+                forms: []
+              },
+            },
+            events: {
```
Comment (egekorkan) [inline index.html:2841]: this event is never emitted. It would be good to give an example

**Review by egekorkan — COMMENTED** [review]
Until Section 9 (I had done until 7 last time)
General feedback:
- Why are some lines wrapped to a certain while others are not?

**Comment by egekorkan** [comment]
Oh I saw #248 too late, sorry about that :( I will do the rest like that

File: index.html [hunk 10]
```diff
@@ -0,0 +1,3643 @@
...
+    </p>
+    <p class="ednote">
+      This specification is implemented at least by the <a href="http://www.thingweb.io/">Eclipse Thingweb</a>
+      project also known as <a href="https://github.com/eclipse/thingweb.node-wot">node-wot</a>, which is considered the reference open source implementation at the moment. Check its <a href="https://github.com/eclipse/thingweb.node-wot"> source code</a>, including <a href="https://github.com/eclipse/thingweb.node-wot/examples">examples</a>.
+    </p>
+  </section>
+
+  <section id="sotd">
+    <p>
+      Implementers need to be aware that this specification is considered unstable. Vendors interested in implementing this specification before it eventually reaches the Candidate Recommendation phase should subscribe to the <a href="https://github.com/w3c/wot-scripting-api">repository</a> and take part in the discussions.
```
[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:143]: This is "standard" warning text. It comes with the status of the doc.

File: index.html [hunk 11]
```diff
@@ -0,0 +1,3643 @@
...
+      </p>
+    </section>
+
+    <section data-dfn-for="InteractionOptions">
+      <h3>
+        The <dfn>InteractionOptions</dfn> dictionary
+      </h3>
+      <p>
+        Holds the interaction options that need to be exposed for application scripts according to the <a>Thing Description</a>.
+      </p>
```
[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:1081]: Makes sense.

File: index.html [hunk 12]
```diff
@@ -0,0 +1,3643 @@
...
+      <p>
+        The <dfn>data</dfn> property if defined, represents additional opaque
+        data that needs to be passed to the interaction.
+      </p>
+    </section>
+
+    <section>
+      <h3>The <dfn>PropertyMap</dfn> type</h3>
+      <p>
+        Represents a map of <a>Property</a> names as strings to a value that the <a>Property</a> can take. It is used as a property bag for interactions that involve multiple <a>Properties</a> at once.
```
[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:1111]: A bit colloquial, yes :), but makes it clear. It's fine to reword.

File: index.html [hunk 13]
```diff
@@ -0,0 +1,3644 @@
...
+          </li>
+          <li>
+            Let |interaction| be the value of ||td||'s |properties|'s
+            |propertyName|.
+          </li>
+          <li>
+            If |option|'s |formIndex| is defined, let |form| be the
+            <a>Form</a> associated with |formIndex| in |interaction|'s |forms|
+            array, otherwise let |form| be the first <a>Form</a> in
+            |interaction|'s |forms| whose |op| is `readproperty`.
```
[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:1144]: This is again standard text also used in other specs. `A's B` means that B is property of A.

[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:1144]: But I agree that adding `value` makes sense, like 

> whose |op|'s value is `readproperty`

File: index.html [hunk 14]
```diff
@@ -0,0 +1,3644 @@
...
+        </ol>
+      </div>
+    </section>
+
+    <section>
+      <h2>ConsumedThing Examples</h2>
+      <p>
+        The next example illustrates how to fetch a <a>TD</a> by URL, create a {{ConsumedThing}}, read metadata (title), read property value, subscribe to property change, subscribe to a WoT event, unsubscribe.
+      </p>
+      <pre class="example" title="Thing Client API example">
```
[the remark this answers is not in the document]
Comment (zolkis) [reply index.html:1670]: The conformance class name is WoT Consumer, indeed.

**Comment by danielpeintner** [comment]
@egekorkan do you want to take a look at the *last* version and check whether all aspects are integrated
https://w3c.github.io/wot-scripting-api/

**Pull request closed without merging**